supervisor/hardware/monitor.py

```diff
@@ -51,39 +55,39 @@ async def unload(self) -> None:
         _LOGGER.info("Stopped Supervisor hardware monitor")
 
     def _udev_events(self, kernel: pyudev.Device):
-        """Incomming events from udev.
-
-        This is inside a observe thread and need pass into our eventloop.
-        """
+        """Incomming events from udev."""
         _LOGGER.debug("Hardware monitor: %s - %s", kernel.action, pformat(kernel))
-        try:
-            udev = pyudev.Devices.from_sys_path(self.context, kernel.sys_path)
-        except pyudev.DeviceNotFoundAtPathError:
-            udev = None
 
-        self.sys_loop.call_soon_threadsafe(
-            self._async_udev_events, kernel.action, kernel, udev
-        )
+        if kernel.action in (UdevKernelAction.UNBIND, UdevKernelAction.BIND):
+            return
+        self.sys_create_task(self._async_udev_events(kernel))
 
-    def _async_udev_events(
-        self, action: str, kernel: pyudev.Device, udev: Optional[pyudev.Device]
-    ):
+    async def _async_udev_events(self, kernel: pyudev.Device):
         """Incomming events from udev into loop."""
         # Update device List
         if not kernel.device_node or self.sys_hardware.helper.hide_virtual_device(
             kernel
         ):
             return
 
+        try:
+            # We get pure Kernel events only inside container.
+            # But udev itself need also time to initialize the device
+            # before we can use it correctly
+            while True:
+                await asyncio.sleep(1)
```

Comment:
I sleep first because I want to give udev a bit of time to update his database and not hammer it agressive